presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1144,6 +1146,16 @@ else if (expression instanceof LongLiteral) {
     {
         List<Set<Expression>> computedGroupingSets = ImmutableList.of(); // empty list = no aggregations
 
+        boolean isGroupingOperationPresent = outputExpressions.stream()
+                .anyMatch(GroupingOperationAnalyzer::containsGroupingOperation);
+
+        if (isGroupingOperationPresent && !node.getGroupBy().isPresent()) {
+            throw new SemanticException(
+                    INVALID_PROCEDURE_ARGUMENTS,
+                    node,
+                    "A GROUPING() operation can only be used with a corresponding GROUPING SET/CUBE/ROLLUP/GROUP BY clause");
```

Comment:
Your suggestion is wrong or I'm misunderstanding. The grouping() function requires a GROUPING SETS/CUBE/ROLLUR/GROUP BY clause. In that sense it's like any other aggregation function, you can't invoke it without doing an aggregation.

Your suggestion sounds like GROUPING SET/CUBE/ROLLUP require a GROUP BY which technically true but is not what we're trying to check for.